cmake/HalideGeneratorHelpers.cmake

```diff
@@ -343,21 +343,27 @@ endfunction()
 
 function(_Halide_target_link_gpu_libs TARGET VISIBILITY)
     if ("${ARGN}" MATCHES "opengl")
-        if (NOT TARGET X11::X11)
+        if (NOT "${ARGN}" MATCHES "egl" AND NOT TARGET X11::X11)
             find_package(X11)
             if (NOT X11_FOUND)
                 message(AUTHOR_WARNING "X11 dependency not found on system.")
             endif ()
         endif ()
-        target_link_libraries(${TARGET} ${VISIBILITY} X11::X11)
+        if (X11_FOUND)
+            target_link_libraries(${TARGET} ${VISIBILITY} X11::X11)
+        endif ()
```

Comment:
This is wrong. `X11_FOUND` is set as a cache variable so X11 will be linked to every target _after_ the first one loads X11 (or if it is loaded previously). I think this is what you meant (it seems you intended to link X11 only if "egl" is not in the target)

```cmake
if (NOT "${ARGN}" MATCHES "egl")
    if (NOT TARGET X11::X11)
        find_package(X11 REQUIRED)
        if (NOT X11_FOUND)
            message(AUTHOR_WARNING "X11 dependency not found on system.")
        endif ()
    endif ()
    target_link_libraries(${TARGET} ${VISIBILITY} X11::X11)
endif ()
```